docs/architecture/api-specs-support.md

```diff
@@ -0,0 +1,35 @@
+# API Specifications Support
+
+## Overview
+
+This document describes current and planned api specification support in Kyma components.
+
+## Currently supported api specification versions and formats
+
+Following table shows currently supported api specification versions and formats in different components.
+
+Last row describes support for extracting api specifications from archive files.
+
+| | Console UI & Compass UI (rendering) | Rafter (storing) |  Compass UI (uploading) | Compass Backend (storing) | Runtime Agent (storing)
+| --- | --- | --- | --- | --- | --- |
+| OpenAPI<br>(YAML, JSON) | all formats, v2 and v3 | no validation / conversion | all versions and formats, validation by property | all versions and formats, no validation | all versions, JSON only, no validation and conversion |
+| OData<br>(XML, JSON in v4) | XML only, all versions converted to v4 on client side |  no validation / conversion |  all versions, XML only, validation by parsing XML file and checking property | all versions and formats, no validation | all versions and formats, no validation and conversion |     
+| AsyncAPI<br>YAML, JSON | all formats, v2 only | validation and conversion to JSON and v2 | all versions and formats, validation by property | all versions and formats, no validation | all versions, JSON only, no validation and conversion |
+| Archives | no | yes (TAR and ZIP) | no | no | no |
+
+## Planned support for api specification versions and formats
+
+Following table shows proposed supported api specification versions and formats in different components.
+Rendering would support only a single version of api specifications.
+Rafter would be responsible for validating and converting api specifications for Console UI.
+Because Compass UI doesn't use Rafter conversion would have to be done in Compass Backend.
+Preferably Rafter's converters and validators should be exposed as services or libraries that could be used by Compass Backend and Compass UI.
```

Comment:
1. Prio one: as a libraries if possible, cause we save specs in our db, and I want to avoid for now additional components in compass cluster.

Reply:
What can be problematic about libraries is that UI won't be able to use them (unless there would be implementations for different languages).